internal/ingress/metric/collectors/socket.go

```diff
@@ -190,7 +190,7 @@ func NewSocketCollector(pod, namespace, class string, metricsPerHost bool, bucke
 				Namespace:   PrometheusNamespace,
 				ConstLabels: constLabels,
 			},
-			[]string{"ingress", "namespace", "status", "service", "canary"},
+			requestTags,
```

Comment:
Wow, I can't believe we had `method` and `path` for histograms all along if the concern was cardinality. Histograms are much more sensitive to cardinality since they have multiple metrics * buckets.

It makes sense to add `method` and `path` to `nginx_ingress_controller_requests`, since these are valuable and each have finite cardinality. LGTM.